runtime/src/main/jni/ObjectManager.cpp

```diff
@@ -297,41 +295,25 @@ void ObjectManager::JSObjectFinalizerStatic(const WeakCallbackInfo<ObjectWeakCal
 }
 
 void ObjectManager::JSObjectFinalizer(Isolate* isolate, ObjectWeakCallbackState* callbackState) {
+    HandleScope handleScope(m_isolate);
     Persistent<Object>* po = callbackState->target;
+    auto jsInstanceInfo = GetJSInstanceInfoFromRuntimeObject(po->Get(m_isolate));
 
-    auto jsInfoIdx = static_cast<int>(MetadataNodeKeys::JsInfo);
-    auto jsInstance = po->Get(m_isolate);
-    auto jsInfo = jsInstance->GetInternalField(jsInfoIdx);
-    if (jsInfo->IsUndefined()) {
-        // Typescript object layout has an object instance as child of the actual registered instance. checking for that
-        auto prototypeObject = jsInstance->GetPrototype().As<Object>();
-        if (!prototypeObject.IsEmpty() && prototypeObject->IsObject()) {
-            DEBUG_WRITE("GetJSInstanceInfo: need to check prototype :%d", prototypeObject->GetIdentityHash());
-            if (IsJsRuntimeObject(prototypeObject)) {
-                jsInfo = prototypeObject->GetInternalField(jsInfoIdx);
-            }
-        }
-    }
-
-    if (jsInfo.IsEmpty() || !jsInfo->IsExternal()) {
-        // The JavaScript instance has been forcefully disconnected from the Java instance.
+    if (!jsInstanceInfo) {
```

Comment:
in case jsInstanceInfo is `nullptr` this check wont pass
please change check to: `jsInstanceInfo != nullptr`